File: adventofcode/2021/1/python3/1.py
def slv(nums, window_width):
    ans = 0
    for i in range(len(nums)-window_width):
        if nums[i] < nums[i+window_width]: ans += 1
    return ans

nums = []
while True:
    try: nums.append(int(input()))
    except EOFError: break
print("Problem1: {}".format(slv(nums, 1)))
print("Problem2: {}".format(slv(nums, 3)))

File: adventofcode/2021/1/c++/1.cpp
#include <bits/stdc++.h>
using namespace std;

int slv(vector<int> a, int n, int window_width) {
    int ans = 0;
    for(int i = 0; i < n-window_width; i++)
        if(a[i] < a[i+window_width]) ans++;
    return ans;
}

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++)
        cin >> a[i];
    cout << "Problem1: " << slv(a, n, 1) << endl;
    cout << "Problem2: " << slv(a, n, 3) << endl;
}